### ISSUE TITLE
URL filling with nulls

### ISSUE THREAD

**Issue by jrosindell** [OneZoom/OZtree#785]
![Screenshot 2024-04-09 at 00 26 16](https://github.com/OneZoom/OZtree/assets/3009279/40490fd7-ef4d-40fc-a0fb-24648cbd8e09)

**Comment by davidebbo** [comment]
@jrosindell I'm not seeing this behavior on Chrome, using the same tour at the same step. My URL looks like https://beta.onezoom.org/life/@biota=93302?otthome=%40_ozid%3D1&tour=%2Ftour%2Fdata.html%2Ffrogs%404#x783,y1066,w1.4031

Maybe there are other factors at play?

**Comment by lentinj** [comment]
I could also believe it only happens if you start the tour in a particular way. Going to a URL like ``https://beta.onezoom.org/life/?tour=%2Ftour%2Fdata.html%2Ffrogs%401`` would start the tour on page-load, and since the original URL doesn't have a pinpoint in, it'd be null.

It's a very neat explanation, but unfortunately the URL above seems to work fine.

**Comment by lentinj** [comment]
Yeah, that's how we're repeating the ``null/``s. Everything before the final pathname part with the ``@`` is treated as the base URL and preserved. Generally this is fine since we only modify the pinpoint at the end of the pathname.

A bodge would be to not record a URL if the pinpoint we're trying to record is ``null``, but this would potentially break the share buttons.

**Comment by lentinj** [comment]
I still can't reproduce this (@hyanwong says that rapid fire "Next"ing can do it, but not here). However, the pull request should at least solve the repeated "null/"s.

I'm still not sure why there would be a state without a pinpoint mind, but the problem is somewhere around here:

https://github.com/OneZoom/OZtree/blob/2ee69002f00a21c963bc346e203ee4ba140f95f1/OZprivate/rawJS/OZTreeModule/src/navigation/record.js#L34-L48

Note that (line 34) refuses to record a URL without a pinpoint, but when in a tour we ignore this state and craft our own based on the current page location. If this URL doesn't have a pinpoint, it'd get passed through regardless.

I don't think the solution is moving this check. We still want to record the current tourstop in the URL, so we can then refresh / share the current URL.
